src/openassetio-python/package/openassetio/managerApi/ManagerInterface.py

```diff
@@ -479,85 +479,104 @@ def finalizedEntityVersion(self, entityRefs, context, hostSession, overrideVersi
+        The traits of the relationship to query.
 
-            a)  getRelatedReferences([ r1 ], [ td1, td2, td3 ])
+        @param toEntityReferences `List[` @fqref{EntityReference} "EntityReference" `]`
+        A list of @ref entity_reference to query the specified
+        relationship for.
 
-            > [ [ r1td1... ], [ r1td2... ], [ r1td3... ] ]
+        @param context Context The calling context.
 
-            b)  getRelatedReferences([ r1, r2, r3 ], [ td1 ])
+        @param hostSession openassetio.managerApi.HostSession The host
+        session that maps to the caller, this should be used for all
+        logging and provides access to the openassetio.managerApi.Host
+        object representing the process that initiated the API session.
 
-            > [ [ r1td1... ], [ r2td1... ], [ r3td1... ] ]
+        @param resultTraitSet `Set[str]` or None, a hint as to what
+        traits the returned entities should have.
+
+        @return `List[List[@fqref{EntityReference} "EntityReference"`]]`
+        A list of references to related entities, for each input
+        reference.
+
+        @note Ensure that your implementation of
+        @fqref{managerApi.ManagerInterface.managementPolicy}
+        "managementPolicy" responds appropriately when queried with
+        relationship trait sets and a read policy to communicate to the
+        host whether or not you are capable of handling queries for
+        those relationships in this method.
+        """
+        return [[] for _ in toEntityReferences]
 
-            c)  getRelatedReferences([ r1, r2, r3 ], [ td1, td2, td3 ])
+    def getWithRelationships(
+        self, relationshipTraitsDatas, toEntityReference, context, hostSession, resultTraitSet=None
+    ):
+        """
+        Returns entity references that are related to the input
+        reference by the relationships defined by a set of traits and
+        their properties.
 
-            > [ [ r1td1... ], [ r2td2... ], [ r3td3... ] ]
+        This is an essential function in this API - as it is widely used
+        to query other entities or organisational structure.
 
         @note The order of entities in the inner lists of matching
-        references will not be considered meaningful, but the outer list
-        should match the input order.
-
-        In summary, if only a single entityRef is provided, it should be
-        assumed that all relationship definitions should be considered
-        for that one entity. If only a single relationship definition is
-        provided, then it should be considered for all supplied entity
-        references. If lists of both are supplied, then they must be the
-        same length, and it should be assumed that it is a 1:1 mapping
-        of a relationship definition to an entity. If this is not the
-        case, ValueErrors should be thrown.
+        references should not be considered meaningful, but the outer
+        list should match the input entity reference order.
```

Comment:
[fix] input relationship order for this method right?

Reply:
Took a different approach [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/915/commits/ad2749f2cafc239c0582fdd26fb14da21ba64e48), which I think is perhaps more accurate?